root/usr/share/ruby/vendor_ruby/discovery/screen/network.rb

```diff
@@ -6,9 +6,9 @@ def screen_network mac, vlan_id, ip = cmdline('fdi.pxip', ''), gw = cmdline('fdi
   l_ip = Newt::Label.new(2, 8, _("Address:"))
   l_gw = Newt::Label.new(2, 10, _("Gateway:"))
   l_dns = Newt::Label.new(2, 12, _("DNS:"))
-  t_ip = Newt::Entry.new(16, 8, ip, 55, Newt::FLAG_SCROLL)
-  t_gw = Newt::Entry.new(16, 10, gw, 55, Newt::FLAG_SCROLL)
-  t_dns = Newt::Entry.new(16, 12, dns, 55, Newt::FLAG_SCROLL)
+  t_ip = Newt::Entry.new(16, 8, ip.to_s, 55, Newt::FLAG_SCROLL)
+  t_gw = Newt::Entry.new(16, 10, gw.to_s, 55, Newt::FLAG_SCROLL)
+  t_dns = Newt::Entry.new(16, 12, dns.to_s, 55, Newt::FLAG_SCROLL)
```

Comment:
The alternative would be to ensure it's a string when wen call `screen_network` in https://github.com/theforeman/foreman-discovery-image/blob/4e8e664ec8cd49016d139c8260bb8e73b89953ac/root/usr/share/ruby/vendor_ruby/discovery/screen/primary_iface.rb#L64-L65

Or we change `detect_ipv4_credentials` (and then probably also `detect_ipv6_credentials`) to return `""` not `nil` when no data was found, but this was intentionally changed to `nil` in 2db98b358f36b294d48f72d8c5400d8d0347eb34